src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health; // Виправлено на health
+    Animal.alive.push(this);
+    this.hidden = false;
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index !== -1) {
+      Animal.alive.splice(index, 1); // Використовуємо splice замість slice
+    }
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      // Виправлено на health
+      this.die();
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
+
+  unHide() {
+    this.hidden = false;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (!(herbivore instanceof Carnivore) && !herbivore.hidden) {
```

Comment:
The condition `!(herbivore instanceof Carnivore)` is incorrect. It should check if `herbivore` is an instance of `Herbivore` instead, i.e., `herbivore instanceof Herbivore`. This ensures that only herbivores can be bitten by carnivores.